app/vlselect/logsql/logsql.go

```diff
@@ -453,16 +480,25 @@ func ProcessLiveTailRequest(ctx context.Context, w http.ResponseWriter, r *http.
 
 var liveTailRequests = metrics.NewCounter(`vl_live_tailing_requests`)
 
-const tailOffsetNsecs = 5e9
+const (
+	tailMaxWindowNsecs   = 30e9 // 30 seconds
+	tailOffsetStartNsecs = 5e9  // 5 seconds
+)
 
 type logRow struct {
 	timestamp int64
 	fields    []logstorage.Field
 }
 
 func sortLogRows(rows []logRow) {
-	sort.SliceStable(rows, func(i, j int) bool {
-		return rows[i].timestamp < rows[j].timestamp
+	slices.SortStableFunc(rows, func(a, b logRow) int {
```

Comment:
This change substitutes two lines of clear code with 8 lines of not so clear code. While `slices.SortStableFunc()` may work faster than `sort.SliceStable()`, this is irrelevant in this context, since it is expected that `sortLogRows` is used for sorting small number of logs.

Also, performance-optimization pull requests must be accompanied with benchmarks, which clearly show performance gains after applying the given performance optimizations.